lib/geometry/polygon.nit

```diff
@@ -0,0 +1,428 @@
+# This file is part of NIT (http://www.nitlanguage.org).
+#
+# Copyright 2014 Romain Chanoir <romain.chanoir@viacesi.fr>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# Convex Polygons manipulations
+module polygon
+
+import points_and_lines
+
+
+# Convex Polygon class
+class ConvexPolygon
+
+	# Vertices of this polygon
+	var points = new Array[Point[Float]]
+
+	# Init this polygon with a list of at least 3 vertices
+	init with_vertices(pts: Array[Point[Float]]) do
+		assert pts.length >= 3
+		points.add_all(pts)
+	end
+
+	# Get an array of x coordinates of the vertices
+	private fun x_coordinates: Array[Float] do
+		var x = new Array[Float]
+		for p in points do
+			x.add(p.x)
+		end
```

Comment:
You can use an array comprehension for this kind of operation:
~~~
return [for p in points do p.x]
~~~